packages/typespec-client-generator-core/src/http.ts

```diff
@@ -552,8 +556,10 @@ export function getCorrespondingMethodParams(
       );
       return diagnostics.wrap([]);
     }
-    return diagnostics.wrap(clientParams.filter((x) => isApiVersion(context, x)));
+    return diagnostics.wrap(existingApiVersion ? [existingApiVersion] : []);
   }
+
+  // 3. To see if the service parameter is subscription parameter that has been elevated to client (only for arm service).
```

Comment:
you could also pass the client initialization parameter to this function. but you also need to pass a method, since we need to find the corresponding http operation parameter. see the `client parameter` test.